src/components/TodoistQuickAdd.tsx

```diff
@@ -0,0 +1,134 @@
+import { useState, useEffect, useRef } from 'react'
+
+const LOADING_TIMEOUT = 5000 // 5sec
+const SCRIPT_ELM_ID = 'todoist-quick-add'
+const TODOIST_HOST = 'todoist.com'
+
+type ShowProps = {
+    content: string
+    todoistHost: string
+    priority?: number
+    date?: string
+    theme?: number
+}
+
+declare global {
+    interface Window {
+        TodoistQuickAddSDK: {
+            show: (parms: ShowProps) => void
+        }
+    }
+}
+
+type Props = {
+    content: string
+    priority?: number
+    date?: string
+    theme?: number
+    todoistHost?: string
+    onAdd?: () => void
+    onClose?: () => void
+    onLoadingSuccess?: () => void
+    onLoadingError?: () => void
+}
+
+type CustomMessageEvent = {
+    data: string
+}
+
+type MessageJSON = {
+    type: string
+}
+
+function TodoistQuickAdd({
+    content,
+    priority,
+    date,
+    theme,
+    todoistHost,
+    onAdd,
+    onClose,
+    onLoadingSuccess,
+    onLoadingError,
+}: Props): null {
+    const [isLoaded, setIsLoaded] = useState<boolean>(false)
+    const loadingTimerToClear = useRef<number | undefined>(undefined)
+
+    // Set up post message listener
+    useEffect(() => {
+        function handleMessageReceived(event: CustomMessageEvent) {
+            const data = event.data
+            if (data && typeof data === 'string' && data.indexOf('Todoist Quick Add SDK')) {
```

Comment:
There is a problem with such comparison pattern: any string message containing `Todoist Quick Add SDK` will pass the test. In bigger app like Twist or 3rd party app there are lots of messages floating around, and if someone e.g. types `Todoist Quick Add SDK` into some input box, it could trigger a message that would pass.

Also see my comment here about possibly sending an object: https://github.com/Doist/todoist-web/pull/2320/files#r531756753

Reply:
@proxi this is already fixed in the latest version 👍